### ISSUE TITLE
Importing namespaced nixosModules

### ISSUE THREAD

**Issue by TakodaS** [snowfallorg/lib#136]
Suppose I really like the palworld server configuration you use in your config and i want to use it in my dotfiles. I setup my snowfall flake as follows:

```nix
#flake.nix
{
  inputs = {
    nixpkgs.url = "github:nixos/nixpkgs/nixos-unstable";

    snowfall-lib = {
      url = "github:snowfallorg/lib";
      inputs.nixpkgs.follows = "nixpkgs";
    };

    jh-config = {
      url = "github:jakehamilton/config";
      inputs.nixpkgs.follows = "nixpkgs";
      inputs.aux-website.url = "nixpkgs"; # I override the flake urls here because nix flake show errors out
      inputs.discord-nord-theme.url = "nixpkgs";
    };
  };

  outputs = inputs:
    inputs.snowfall-lib.mkFlake {
      inherit inputs;
      src = ./.;
      systems.modules.nixos = with inputs; [
        jh-config.nixosModules."services/palworld"
      ];
    };
}

#systems/x86_64-linux/test-system/default.nix
{
  config,
  namespace, # The namespace used for your flake, defaulting to "internal" if not set.
  ...
}: {
  config.services.palworld.enable = true;
}
```

This fails with the error
`error: The option plusultra.services.steam does not exist.`
because this dependent option is not loaded. To fix this, I could add ` jh-config.nixosModules."services/steam" to my module imports but this might depend on other internally defined options and would mean I'd have to manually keep adding modules until the dependencies are satisfied. On the other hand, i can see how polluting the global namespace might have nasty consequences down the line. Is there a smarter way I'm missing? Many thanks!

**Comment by jameslounds** [comment]
I think this is an issue with how that specific module is written rather than with snowfall itself. [This line](https://github.com/jakehamilton/config/blob/6158f53f916dc9522068aee3fdf7e14907045352/modules/nixos/services/palworld/default.nix#L77) makes explicit reference to `config.plusultra.services.steam` so if you use this module, you need to have `config.plusultra.services.steam` as an option in your config, since it's not declared by that file.

**Comment by TakodaS** [comment]
Right, but this implies you have to manually handle dependencies which is certainly not advisable. Considering that any slight modification you would want to make would require some overrides and whatnot, it's probably just better to copy/paste into your own snowfall configuration.

Closing because external nixos module dependency management is out of scope.
